src/python/pants/init/engine_initializer.py

```diff
@@ -98,7 +98,13 @@ class GraphSession:
     goal_map: Any
 
     # NB: Keep this in sync with the method `run_goal_rules`.
-    goal_param_types: ClassVar[tuple[type, ...]] = (Specs, Console, Workspace, EnvironmentName)
+    goal_param_types: ClassVar[tuple[type, ...]] = (
+        Specs,
+        Console,
+        Workspace,
+        EnvironmentName,
+        CurrentExecutingGoal,
+    )
```

Comment:
@stuhood I found this spot for it. Feels so much more appropriate than in session values! ;)

Reply:
Ugh. OK. Let's do that, can sync on slack.